config/routes.rb

```diff
@@ -62,6 +62,10 @@
     resources :products, only: [:index] do
       resources :product_accessories, only: [:index, :create, :destroy], path: "accessories"
       resources :training_requests, only: [:new, :create] if SettingsHelper.feature_on?(:training_requests)
+
+      get "notifications", to: "product_notifications#index", as: "notifications"
+      get "notifications/edit", to: "product_notifications#edit", as: "edit_notifications"
+      post "notifications", to: "product_notifications#update"
     end
```

Comment:
How about we use resource-based routes for these? It’s a brand-new resource 😄

Reply:
I started that way and it wasn't working the way I wanted it to. `edit`/`update` need IDs. Conceptually, I could see each of these fields being their own "resource", but we're bulk editing them on a single page.

I could use `new`/`create` to get the routes working, but that didn't feel right.

What about 
```ruby
     resources :product_notifications, only: :index, path: "notifications", as: "notifications" do
        get :edit, on: :collection
        post :update, on: :collection
      end
```

What do you think?

Reply:
I was thinking that we could use a singleton resource here, which doesn’t need ids for edit and update. Given that a product has one set of notifications it may send, it seems that the following would fit the bill:

```
resource :product_notifications, only: [:show, :edit, :update]
```

Since we’re editing all the notifications on a single page, if we treat “the set of notifications tied to this product” as the resource, I think this gets us exactly what we want.

Reply:
Ah, good call. I forgot about the singleton `resource` option.